### DISCUSSION TITLE
Wazuh OSSEC agent cant communicate after so-allow rule added

### DISCUSSION THREAD

**Discussion by RobertClum** [Security-Onion-Solutions/securityonion#6886] [Unsupported Versions] [answered]
Exact version: 2.3.91

Install source. Did you install from our Security Onion ISO image or did you perform a network installation? Both

If network install, did you install from CentOS 7 or Ubuntu 18.04? initially Ubuntu 20.04, than did fresh install with Security Onion ISO 2.3.91

Install type = standalone

Does so-status show all services running? yes

Do you get any failures when you run salt-call state.highstate from the node? No 

Explain your issue:
I initially had the issue on my ubuntu 20.04 install that I did. Spent awhile troubleshooting, but could not determine the issue, so I decided to do a fresh install with the official Security Onion ISO “securityonion-2.3.91.iso”. No issues were encountered with the installation and OSSEC logs were being ingested from the actual SO host.  When the new install came up I verified the logs were coming in via the Hunt interface. I was reviewing alerts in the alerts dashboard and notice an alert for “Ossec agent disconnected”. The only configuration change I did around that time was a so-allow rule to allow for logbeats to ingest windows host logs, it could be not related but figured I would mention it since its the only thing I did after install.

Have you run soup to ensure that you’re on the latest version? 
Yes, 2.3.91

Check the FAQ.
Didn't find relevant FAQ

Search the Community Support forum.
Yes, could not find anything that fit my exact issue

Search the documentation and support forums of the tools contained within Security Onion: Tools

Check log files in /opt/so/log/ or other locations for any errors or possible clues: I reviewed logs below

Setup /root/sosetup.log - Nothing noteworhty found
Suricata /opt/so/log/suricata/suricata.log - Nothing noteworhty found
Zeek /nsm/zeek/logs/current/ - Nothing noteworhty found
Elasticsearch /opt/so/log/elasticsearch/<hostname>.log - Nothing noteworhty found
Kibana /opt/so/log/kibana/kibana.log - Nothing noteworhty found
Logstash /opt/so/log/logstash/logstash.log - alot of these below, but I dont think are relavent to the issue.
[2022-01-15T18:45:19,447][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"so-elasticsearch-2022.01.15"
Elastalert /opt/so/log/elastalert/elastalert_stderr.log - didnt exsist

Additional Checked Logs:
/var/log/firewalld - had some errors (attached)
[SO-Firewall-logs.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7875912/SO-Firewall-logs.txt)
/var/ossec/logs/ossec.log - had some errors(attached)
[SO-Wazuh-Logs.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7875911/SO-Wazuh-Logs.txt)
/var/ossec/logs/ossec.log - inside so-wazuh container (attached)
[SO-Wazuh-OSSEC-container-Logs.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7875908/SO-Wazuh-OSSEC-container-Logs.txt)

Are you able to duplicate the problem on a fresh Security Onion installation? Yes

Check the Known Issues to see if this is a known issue that we are working on. - I did not see a relavent issue in known issues when I looked.

**Comment by RobertClum** [comment]
Not sure why, but I looked this morning and the agent is checking in without errors. I also was able to authenticate and ingest information from a windows 10 host as well. Currently it appears to have resolve itself, I will post back if the issue returns.

**Reply by RobertClum** [reply]
Looks like the issue returned, the agent is able to register with the server, but all the connections on port 1514 are being reset by the SO server.

**Reply by kl3ss** [reply]
@RobertClum Are your agents trying to connect to the Wazuh Manager /  SO Server using TCP or UDP on port 1514?

**Reply by RobertClum** [reply]
@kl3ss I've tried both tcp and udp and still failing.

**Comment by dougburks** [comment]
Is your Security Onion box using a static IP address or DHCP?

**Reply by RobertClum** [reply]
@dougburks 

It is using a static IP.

It's able to register the Wazuh agent no problem, but it cant ingest the data on port 1514 tcp or udp.

I am ingesting logs with winlogbeats into elastic with no issue.

**Comment by RobertClum** [comment, marked as answer]
Went into the container and I noticed wazuh-manager service was not running, tried a service start but received error, see below.

```
root@securityonion-wazuh-manager:/var/ossec/bin# service --status-all
 [ ? ]  hwclock.sh
 [ - ]  postfix
 [ - ]  procps
 [ + ]  wazuh-api
 [ - ]  wazuh-manager
```


```
root@securityonion-wazuh-manager:/var/ossec/bin# service wazuh-manager status  
wazuh-clusterd not running...
wazuh-modulesd not running...
ossec-monitord not running...
ossec-logcollector not running...
ossec-remoted not running...
ossec-syscheckd not running...
ossec-analysisd not running...
ossec-maild not running...
ossec-execd not running...
wazuh-db not running...
ossec-authd is running...
ossec-agentlessd not running...
ossec-integratord not running...
ossec-dbd not running...
ossec-csyslogd not running...
```

received this error when starting service

```
root@securityonion-wazuh-manager:/var/ossec/bin# service wazuh-manager start 
2022/01/19 02:55:40 ossec-analysisd: ERROR: Duplicate rule ID:510
2022/01/19 02:55:40 ossec-analysisd: CRITICAL: (1220): Error loading the rules: 'etc/rules/local_rules.xml'.
ossec-analysisd: Configuration error. Exiting
root@securityonion-wazuh-manager:/var/ossec/bin#
``` 

Based on the Critical error; I figured maybe the rules I added in the local rules file wazuh didn't like, so I removed them from etc/rules/local_rules.xml and then was able to start the service back up.

Now I am able to ingest the wazuh agent logs on the SO Server, windows 10 machine and MacOS.

Hope this helps someone in the future!
